lib/protobuf-c.cmake

```
@@ -0,0 +1,61 @@
+# Compile protobuf-c
+include(CheckLibraryExists)
+include(ExternalProject)
+
+if(BUILD_ONLY_DOCS)
+  # skip
+elseif(NOT BUILD_PROTOBUFC_LIB)
+  find_package(protobuf-c MODULE REQUIRED)
+  message("Found protobuf-c library: ${protobuf-c_LIBRARIES}")
+else()
+  # Download and Compile protobuf-c library at compile time using ExternalProject
+  # (e.g. when running `cmake --build` or `make`)
+  set(LIBPROTOBUFC_INSTALL_ROOT "${CMAKE_CURRENT_BINARY_DIR}/lib")
+  set(LIBPROTOBUFC_INSTALL_DIR "${LIBPROTOBUFC_INSTALL_ROOT}/protobuf-c")
+
+  if(BUILD_SHARED_LIBS)
+    set(configure_args "--enable-shared" "--disable-static" "--disable-protoc")
+  else()
+    set(configure_args "--enable-static" "--disable-shared" "--disable-protoc")
+  endif()
```

This is what we need to add to get the build working for cross-compiling.

```suggestion
  endif()
  # let `./configure` know if we're cross-compiling
  list(APPEND configure_args "--host=${CMAKE_LIBRARY_ARCHITECTURE}")
```